includes/class-bcf-google-fonts-compatibility.php

```diff
@@ -0,0 +1,707 @@
+				self::$instance = new self();
+			}
+			return self::$instance;
+		}
+
+		/**
+		 * Constructor
+		 *
+		 * @return void
+		 * @since x.x.x
+		 */
+		public function __construct() {
+			if ( empty( $_GET['page'] ) || BSF_CUSTOM_FONTS_ADMIN_PAGE !== $_GET['page'] ) { //phpcs:ignore WordPress.Security.NonceVerification.Recommended
+				return;
+			}
+
+			$bcf_filesystem    = bcf_filesystem();
+			$fonts_folder_path = $this->get_fonts_folder();
+
+			if ( file_exists( $fonts_folder_path ) ) {
+				$bcf_filesystem->delete( $fonts_folder_path, true, 'd' );
+			}
+
+			self::delete_all_theme_font_family();
+			add_action( 'admin_init', array( $this, 'update_fse_theme_json' ) );
+		}
+
+		/**
+		 * Get the base path.
+		 *
+		 * @access public
+		 * @since x.x.x
+		 * @return string
+		 */
+		public function get_base_path() {
+			if ( ! $this->base_path ) {
+				$this->base_path = apply_filters( 'bcf_fonts_base_path', $this->get_filesystem()->wp_content_dir() );
+			}
+			return $this->base_path;
+		}
+
+		/**
+		 * Get the base URL.
+		 *
+		 * @access public
+		 * @since x.x.x
+		 * @return string
+		 */
+		public function get_base_url() {
+			if ( ! $this->base_url ) {
+				$this->base_url = apply_filters( 'bcf_fonts_base_url', content_url() );
+			}
+			return $this->base_url;
+		}
+
+		/**
+		 * Save Google Fonts to locally.
+		 *
+		 * @return void
+		 * @since x.x.x
+		 */
+		public function update_fse_theme_json() {
+
+			$all_fonts = Bsf_Custom_Fonts_Render::get_instance()->get_existing_font_posts();
+
+			if ( empty( $all_fonts ) || ! is_array( $all_fonts ) ) {
+				return;
+			}
+
+			foreach ( $all_fonts as $font => $id ) {
+				$new_font_faces = array();
+				$font_family    = get_the_title( $id );
+				$font_slug      = $this->get_font_slug( $font_family );
+				$font_data      = get_post_meta( $id, 'fonts-data', true );
+
+				if ( ! empty( $font_data['variations'] ) ) {
+					foreach ( $font_data['variations'] as $variation ) {
+						$font_weight = ! empty( $variation['font_weight'] ) ? $variation['font_weight'] : '';
+						$font_style  = ! empty( $variation['font_style'] ) ? $variation['font_style'] : 'normal';
+
+						$final_font_files = $this->get_fonts_file_url( $font_family, $font_weight, $font_style, $id, true );
+
+						// Convert paths to URLs.
+						foreach ( $final_font_files as $remote => $local ) {
+							$final_font_files[ $remote ] = str_replace(
+								$this->get_base_path(),
+								$this->get_base_url(),
+								$local
+							);
+						}
+
+						// Add each variant as one font face.
+						$new_font_faces[] = array(
+							'fontFamily'  => $font_family,
+							'fontStretch' => '',
+							'fontStyle'   => $font_style,
+							'fontWeight'  => $font_weight,
+							'src'         => $final_font_files,
+						);
+					}
+					$this->add_or_update_theme_font_faces( $font_family, $font_slug, $new_font_faces );
+				}
+			}
+		}
+
+		/**
+		 * Save Google Fonts to locally.
+		 *
```

Comment:
@params are not added to the function @imnavanath

Reply:
DONE